src/parseTools.js

```diff
@@ -1043,3 +1043,12 @@ function getPerformanceNow() {
 function implicitSelf() {
   return ENVIRONMENT.includes('node') ? 'self.' : '';
 }
+
+function ENVIRONMENT_IS_MAIN_THREAD() {
+  var envs = [];
+  if (USE_PTHREADS) envs.push('ENVIRONMENT_IS_PTHREAD');
+  if (WASM_WORKERS) envs.push('ENVIRONMENT_IS_WASM_WORKER');
+  if (AUDIO_WORKLET) envs.push('ENVIRONMENT_IS_AUDIO_WORKLET');
+  if (envs.length == 0) return 'true';
```

Comment:
How about `assert(anvs.length, 'ENVIRONMENT_IS_MAIN_THREAD called in single threaded build')` there instead since if you call this function are when not targetting threads that is probably sign that something else is wrong, no?

Also, how about just calling this `isMainThread()`.  We don't have any other upper-case-only functions here.

Reply:
I don't think `assert(envs.length > 0);` would work, because (if we think this would be a general function) even single-threaded builds could parse library_x functions that could contain a macro like that?

Or if your point is that we should by design restrict the macro to be only called from library_pthread.js, then I could also move the implementation into start of library_pthread.js so it only is applicable there?

Reply:
I think that any code calling code should be wrapped in `#if USE_PTHREADS` or `#if SHARED_MEMORY`

Reply:
i.e. there is no point in using the `isMainThread` macro in a single threaded build, and if you do use it that is probably sign that something is wrong and need to wrap you code in `#if SHARED_MEMORY`.

But also yes, until this helper is more generally useful, limiting its scope to this file would be good too.

Reply:
> there is no point in using the `isMainThread` macro in a single threaded build

Coming back to this PR, and had a look again at this. I'm not sure that is the right angle. I mean, it is not a piece of code that decides what the build configuration is, but rather the user of the code decides the build configuration. We rarely have "single thread only" code files: the library_pthread_stub.js file is the only one that is single thread only, and it is a special exception.

Every other JS library file is built either as singlethreaded or a multithreaded configuration. So in any of those files, having a bare `if ({{{ ENVIRONMENT_IS_MAIN_THREAD() }}}) {` check can be useful. Needing to gate such checks additionally behind `#if SHARED_MEMORY` would make the use of such macro more wieldy. So I think the current use is ideal.